core/ledger/kvledger/kv_ledger.go

```diff
@@ -480,6 +502,14 @@ func (l *kvLedger) CommitLegacy(pvtdataAndBlock *ledger.BlockAndPvtData, commitO
 	elapsedBlockProcessing := time.Since(startBlockProcessing)
 
 	startBlockstorageAndPvtdataCommit := time.Now()
+
+	logger.Debugf("[%s] Committing pvtdata and block [%d] to storage", l.ledgerID, blockNo)
+	// need to unlock snapshotGenerationRWLock before each return depending on code paths
+	l.snapshotGenerationRWLock.Lock()
+	defer l.snapshotGenerationRWLock.Unlock()
```

Comment:
Not sure what you meant by this code comment.

Also, can you make following two minor changes 
- Move this lock acquisition in the commitLegacy function before calling the commit function (just to avoid side-effects such as here, this may cause `elapsedBlockstorageAndPvtdataCommit` to be a very high value, while the commit may actually be waiting for background snapshot generation to finish

- Keep some flag if a background snapshot generation is ON and print a log message (like - commit waiting on .... in case a user may wonder what's happening with the commit). Feel free to tackle this in a separate PR, if you will